src/connection/server_support.c

```diff
@@ -1797,6 +1799,9 @@ css_pack_message_to_master (const char *server_name)
   char pid_string[16];
   std::string message;
 
+  SERVER_TYPE type = get_server_type ();
+  assert (type == SERVER_TYPE_TRANSACTION || type == SERVER_TYPE_PAGE);
```

Comment:
How about move this assert in the set_server_type()?

Reply:
then it needs refactoring I think. 

The g_server_type is not only changed using the set_server_type() function, but there are also direct modifications to it. That is why the assert statement is placed here.

Reply:
only one place init_server_type().

Reply:
OK, then I will move this assertion to set_server_type ()
In init_server_type (), it also add assertion (at get_server_type_from_config ()) when setting the g_server_type.

Reply:
@joohok I agreed.